### PR TITLE
Contact Tracing Report

### PR DESCRIPTION
**Pull request by mth-1012**
# General info

**Issue number**: #15 #19 and other small task issues

**Task description**: 

 - Add routes for contact tracing report (CTR): adding, fetching and updating
 - Add a new tab for a Patient to submit a CTR
 - Add a new tab for HealthOfficial to check on the CTR of a certain Patient
 - Allow HealthOfficial to send a notification and email to the user(s) reported in the CTR
 - In case the email does not match any User in the database, send an invitation email instead
 
# Testing

**Test coverage**:

- Some integration tests were written for new backend functions
- All features were tested manually with Postman in the backend
- Frontend testing are being done...

# Additional notes

Review carefully
Report if found bugs/crashes

### CODE REVIEW

**Comment by p-carrier** [comment]
@mth-1012 could you fix the merge conflicts

File: domain/notification.js [hunk 1]
```diff
@@ -20,6 +22,29 @@ class Notification {
     return await this.notificationRepository.getNotification(this.notificationId);
   }
 
+  async sendNewNotificationEmail(userEmail) {
```
Comment (AC159) [inline domain/notification.js:25]: You shouldn't copy/paste the sendEmail function. There should be only one function that sends an email that accepts an email and a message as a parameter. Inside these kind of functions that send emails, we should call that one function that will actually send the email

File: domain/user.js [hunk 2]
```diff
@@ -25,6 +27,30 @@ class User {
...
     return await this.userRepository.fetchAllNotifications(this.id.getId());
   }
+
+  async sendNonUserEmail(userEmail, inviteMessage) {
+    const transporter = nodemailer.createTransport({
+      host: 'smtp.gmail.com',
+      port: 587,
+      secure: false, // true for 465, false for other ports
+      auth: {
+        user: process.env.COVICARE_EMAIL, // generated ethereal user
```
Comment (AC159) [inline domain/user.js:37]: Same as before, inside this function we should call a single function that handles the sending of emails

File: frontend/src/components/Tabs/AdminTabs/PatientList.js [hunk 3]
```diff
@@ -30,7 +30,7 @@ function PatientList(props) {
         setPatientList(response.data.data);
         console.log(response.data.data)
     } catch (error) {
-        console.log(error);
+        //console.log(error);
```
Comment (AC159) [inline frontend/src/components/Tabs/AdminTabs/PatientList.js:33]: Uncomment the console log

File: frontend/src/components/Tabs/PatientSymptoms/ContactTrace.js [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+import moment from "moment";
+import Modal from "react-bootstrap/Modal";
+import styles from "./ContactTrace.module.css";
+import {Accordion, Button} from "react-bootstrap";
+
+function ContactTrace(props) {
+    let {currentUser} = useAuth();
+    let [locationDescription, setLocationDescription] = useState('');
+    let [emailList, setEmailList] = useState([{email: ''}]);
+    // let [emailListDB, setEmailListDB] = useState([{email: ''}]);
```
Comment (AC159) [inline frontend/src/components/Tabs/PatientSymptoms/ContactTrace.js:14]: If this state is not used it should be removed and not commented

File: frontend/src/components/Tabs/PatientSymptoms/ContactTrace.js [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+                                }
+                                if (!value || key === '_id' || key === 'patientUid' || key === 'timeStamp' || value.length === 0) return null;
+
+                                if (Array.isArray(value)) {
+                                    return <div key={index}><strong>{key}</strong>: <ul>{value.map((element, i) => <li
+                                        key={i}>{element}</li>)}</ul></div>;
+                                } else return <div key={index}><strong>{key}</strong>: {value}</div>;
+                            })}
+                            {/*<Button variant="info" onClick={() => {*/}
+                            {/*    showModalForTracingFormUpdate(index);*/}
```
Comment (AC159) [inline frontend/src/components/Tabs/PatientSymptoms/ContactTrace.js:135]: Remove commented code

**Review by AC159 — APPROVED** [review]
- I think it's good to merge

**Review by p-carrier — APPROVED** [review]
For me everything is fine. I think we could merge

**Pull request merged**